app/src/test/java/ch/epfl/sweng/erpa/services/GCP/GCPRemoteServicesProviderTest.java

```diff
@@ -0,0 +1,126 @@
+    public void testProviderName() {
+        grsp = new TestyGCPRSP(false);
+        assertTrue(grsp.getFriendlyProviderName().contains("Google"));
+        assertTrue(grsp.getFriendlyProviderDescription().contains("ERPA"));
+        assertTrue(grsp.getFriendlyProviderDescription().contains("Google"));
+    }
+
+    @Test
+    public void testCallAndReturn() throws IOException, ServerException {
+        grsp = new TestyGCPRSP(false);
+        service = grsp.getRetrofit().create(TestService.class);
+        Optional<ResponseBody> res = grsp.callAndReturnOptional(service.doCall());
+        assertTrue(res.isPresent());
+        String resString = res.get().string();
+        assertFalse(resString.isEmpty());
+    }
+
+    @Test
+    public void testEmptyOn404() throws IOException, ServerException {
+        grsp = new TestyGCPRSP(true);
+        service = grsp.getRetrofit().create(TestService.class);
+        Call<?> call = service.doCall();
+        assertFalse(grsp.callAndReturnOptional(call).isPresent());
+    }
+
+    @Test(expected = NoSuchElementException.class)
+    public void testExceptionOn404() throws IOException, ServerException {
+        grsp = new TestyGCPRSP(true);
+        service = grsp.getRetrofit().create(TestService.class);
+        Call<ResponseBody> call = service.doCall();
+        grsp.executeAndThrowOn404(call);
+    }
+
+    @Test
+    public void testAsyncRequestSuccess() {
+        grsp = new TestyGCPRSP(false);
+        Pair<Integer,Integer> p = testUpAsync();
+        assertEquals(1,p.getFirst().intValue());
+    }
+
+    @Test
+    public void testAsyncRequestFailure() {
+        grsp = new TestyGCPRSP(true);
+        Pair<Integer,Integer> p = testUpAsync();
+        assertEquals(1,p.getSecond().intValue());
+    }
+
+    private Pair<Integer,Integer> testUpAsync() {
+        service = grsp.getRetrofit().create(TestService.class);
+        Call<Object> call = mock(Call.class);
+        ArgumentCaptor<Callback<Object>> argCaptor = ArgumentCaptor.forClass(Callback.class);
+        AtomicInteger exCount = new AtomicInteger();
+        AtomicInteger successCount = new AtomicInteger();
+        grsp.executeRequest(call, resp -> {
+            successCount.getAndIncrement();
+        }, exception -> {
+            exCount.getAndIncrement();
+        });
+        verify(call, times(1)).enqueue(argCaptor.capture());
+        argCaptor.getValue().onFailure(call,new IOException());
+        argCaptor.getValue().onResponse(call,success(new Object()));
+        return new Pair<>(successCount.get(),exCount.get());
+    }
+
+    private static interface TestService {
+        @GET("/json")
+        Call<ResponseBody> doCall();
+    }
+
+    private static class TestyGCPRSP extends GCPRemoteServicesProvider {
+
+        @NonNull @Override public Retrofit getRetrofit() {
+            return retrofit;
+        }
+
+        @NonNull private Retrofit retrofit = new Retrofit.Builder().baseUrl("http://irrelevant.com").build();
```

Comment:
Can you mock retrofit instead? I don't like a dependency on an external website. If you must use a live system, please use a well-known network with less probability of going down.